### PR TITLE
APIC Mailtrap credentials need removing

### PR DESCRIPTION
**Pull request by Kinshuk1993**
APIC Mailtrap credentials need removing

### CODE REVIEW

File: products/bash/configure-apic-v10.sh [hunk 1]
```diff
@@ -83,6 +83,17 @@ if [[ "$STAGING_AUTHS" == "" || "$STAGING_AUTHS" == "null" ]]; then
 else
   REPO="cp.stg.icr.io"
 fi
+
+if [[ $(oc get secret cp4i-demo-apic-smtp-secret -n "$NAMESPACE") ]]; then
+  MAIL_SERVER_HOST=$(oc get secret cp4i-demo-apic-smtp-secret -n "$NAMESPACE" -o json | jq -r '.data.mailServerHost' | base64 --decode)
+  MAIL_SERVER_PORT=$(oc get secret cp4i-demo-apic-smtp-secret -n "$NAMESPACE" -o json | jq -r '.data.mailServerPort' | base64 --decode)
+  MAIL_SERVER_USERNAME=$(oc get secret cp4i-demo-apic-smtp-secret -n "$NAMESPACE" -o json | jq -r '.data.username' | base64 --decode)
+  MAIL_SERVER_PASSWORD=$(oc get secret cp4i-demo-apic-smtp-secret -n "$NAMESPACE" -o json | jq -r '.data.password' | base64 --decode)
```
Comment (DanRoseus) [inline products/bash/configure-apic-v10.sh:91]: I think we should use the same names as originally used in the CR:
```suggestion
  MAIL_SERVER_USERNAME=$(oc get secret cp4i-demo-apic-smtp-secret -n "$NAMESPACE" -o json | jq -r '.data.mailServerUsername' | base64 --decode)
  MAIL_SERVER_PASSWORD=$(oc get secret cp4i-demo-apic-smtp-secret -n "$NAMESPACE" -o json | jq -r '.data.mailServerPassword' | base64 --decode)
```

File: products/bash/configure-apic-v10.sh [hunk 2]
```diff
@@ -83,6 +83,17 @@ if [[ "$STAGING_AUTHS" == "" || "$STAGING_AUTHS" == "null" ]]; then
...
+  MAIL_SERVER_EMAIL_ADDRESS=$(oc get secret cp4i-demo-apic-smtp-secret -n "$NAMESPACE" -o json | jq -r '.data.emailAddress' | base64 --decode)
```
Comment (DanRoseus) [inline products/bash/configure-apic-v10.sh:92]:
```suggestion
  PORG_ADMIN_EMAIL=$(oc get secret cp4i-demo-apic-smtp-secret -n "$NAMESPACE" -o json | jq -r '.data.emailAddress' | base64 --decode)
```

**Review by DanRoseus — CHANGES_REQUESTED** [review]

File: products/bash/configure-apic-v10.sh (hunk 2 above)
Comment (Kinshuk1993) [reply products/bash/configure-apic-v10.sh]: Done

File: products/bash/configure-apic-v10.sh (hunk 1 above)
Comment (Kinshuk1993) [reply products/bash/configure-apic-v10.sh:91]: When creating a secret without giving a `username` and `password` in it, it gives an error saying `username` and `password` are required fields:
```bash
The Secret "cp4i-demo-apic-smtp-secret-1" is invalid: 
* data[%s][username]: Required value
* data[%s][password]: Required value
```

</br>

The secret yaml:
```yaml
cat <<EOF | oc apply --namespace ${NAMESPACE} -f -
apiVersion: v1
kind: Secret
metadata:
  name: cp4i-demo-apic-smtp-secret-1
type: kubernetes.io/basic-auth
stringData:
  emailAddress: "your@email.address"
  mailServerHost: "smtp.mailtrap.io"
  mailServerPort: "2525"
  mailServerUsername: "your-username"
  mailServerPassword: "your-password123"
EOF
```

Comment (DanRoseus) [reply products/bash/configure-apic-v10.sh:91]: The type should be:
```
type: Opaque
```
See [these docs](https://kubernetes.io/docs/concepts/configuration/secret/#secret-types)

image

When specifying basic auth as the type you need username/password. For Opaque you can have any you like.

Comment (Kinshuk1993) [reply products/bash/configure-apic-v10.sh]: Done

**Pull request merged**